### ISSUE TITLE
We need some signifier of progress through the installer flow

### ISSUE THREAD

**Issue by local-optimum** [canonical/ubuntu-desktop-installer#1348]
![image](https://user-images.githubusercontent.com/55029393/215432565-3344d3ce-f6b0-41e6-8385-5bd51ba4ddf8.png)

Ubiquity included a number of dots to visualise to the user how far through the process they are. Whilst the dot approach is not necessary it would be more user friendly to have a visual presentation of progress through this UI even if not 100% accurate for all use-cases.

Windows 10 has 'chapter headings' throughout the flow to signify this as an alternative example 
![image](https://user-images.githubusercontent.com/55029393/215433077-e8563e6c-6434-46a7-97aa-2178100efe56.png)
![image](https://user-images.githubusercontent.com/55029393/215433137-9e1d6de5-ad3c-4b2a-b60f-9c61f1e8daa7.png)

@elioqoshi it might be good to experiment with mockups of a preferred presentation

**Comment by elioqoshi** [comment]
I agree that information on progress would be useful. Here are a few explorations I did, including what we originally had in Ubiquity:

![Mockups](https://user-images.githubusercontent.com/5436686/215442560-7a990eae-87ae-4e8f-aa73-d20d513fa6bc.png)

I feel the Windows approach is a bit too much. Due to being unable to see non-active menu progress items, it looks like the labels are floating. I'd recommend A or B

**Comment by elioqoshi** [comment]
With the new merged title approach JP worked on:

<img width="1034" alt="image" src="https://user-images.githubusercontent.com/5436686/215447462-131ee085-d3af-4cab-a54c-be632eb35fca.png">

**Comment by elioqoshi** [comment]
An alternative could be also this without losing too much height screen estate:
<img width="1105" alt="image" src="https://user-images.githubusercontent.com/5436686/215455474-d8e473e8-b635-42d2-a0e8-f447b3a0544c.png">

**Comment by local-optimum** [comment]
I like both, if the dots were in line with the buttons on the bottom then I doubt we'd lose much screen estate either @jpnurmi what is the most logistically feasible to implement and how many 'dots'/stages do we need?

**Comment by jpnurmi** [comment]
I recall there are some screens that show additional status/info/warnings at the bottom left that we need to avoid overlapping with, but otherwise putting the dots there at the bottom is trivial. Deciding on the number of steps is harder. :)

I need to take a closer look at how Ubiquity does it because there are some special pages where the steps might not make sense (_Turn off RST/BitLocker_) e.g. _Try or install_ is going to be only shown on the first launch. Maybe all partitioning related could be considered one step because the offered pages depend on your selections and available disks.

**Comment by local-optimum** [comment]
Your approach sounds good to me, I think we should largely echo what was in Ubiquity and not worry about the dots being of equal length in terms of steps. It should be based on the number of installation 'sections'

**Comment by jpnurmi** [comment]
It looks nice when the page indicator is aligned with the buttons but there are a couple of pages that have some info/status/warning labels in the bottom left corner. Does it look silly if we move the page indicator down closer to the window edge?

![image](https://user-images.githubusercontent.com/140617/219975675-c1278363-4e41-4e11-a79d-363544ffba36.png)

![image](https://user-images.githubusercontent.com/140617/219975688-f81117bd-9505-479a-a6ef-ccbd343efbec.png)

![image](https://user-images.githubusercontent.com/140617/219975694-7c38f82f-3013-4418-bd99-6e41e506e386.png)

![image](https://user-images.githubusercontent.com/140617/219975701-2448d133-cbc9-4021-a299-f0b3d4147b8e.png)

![image](https://user-images.githubusercontent.com/140617/219975709-b0620cc1-8dff-48db-93d2-d28104a5caea.png)

![image](https://user-images.githubusercontent.com/140617/219975717-c20dcdb6-92af-4c80-a715-e71402ab1c6b.png)

![image](https://user-images.githubusercontent.com/140617/219975736-51336cf8-2616-4b56-acca-a7256af43124.png)

![image](https://user-images.githubusercontent.com/140617/219975747-405c7678-8d5a-41a8-ac9b-fedc2b5b400b.png)

![image](https://user-images.githubusercontent.com/140617/219975793-7755684c-3610-4682-986d-d81b9bae2ca8.png)

![image](https://user-images.githubusercontent.com/140617/219975799-0ed3d631-22d6-4650-a234-5096d1cc3686.png)

![image](https://user-images.githubusercontent.com/140617/219975866-0bc20f43-60e1-4383-a1b0-193739bb1201.png)

![image](https://user-images.githubusercontent.com/140617/219975836-3b854934-6431-44b9-a60d-efa42cf8e2a4.png)

**Comment by Feichtmeier** [comment]
Duplicate of #135 :)

**Comment by Feichtmeier** [comment]
Maybe there is one more option.

Since the YaruWindowTitleBar includes a title which describes the *content* of the page, eventually it could be followed by a simple - 3/20 page indication to show *where* the page is in the page flow ?

**Comment by local-optimum** [comment]
We did discuss page number style progress indicators, but since different sections of the installer include more pages depending on selection it felt better to go with the abstract 'dots' that simply represent your current 'stage' rather than a specific number of screens. @madsrh I agree that it can be misleading sometimes (hence the need for abstractness in the visualisation) but we also don't want users to not have a rough sense of how far they are through the setup. I agree with @elioqoshi's comment that the current dots should be inline with the continue buttons and more compact. If folk want to argue strongly for not having progress indicators I'm also happy to hear.

**Comment by elioqoshi** [comment]
> Since the YaruWindowTitleBar includes a title which describes the content of the page, eventually it could be followed by a simple - 3/20 page indication to show where the page is in the page flow ?

I don't think that's ideal due to cognitive overload. We do not want to prioritize telling the user exactly where they are, we just want to give them an idea at what position they are in, hence visualizing that is more appropriate than a label. What @local-optimum said as well.

I do think @madsrh seems ideal if that's possible. The snack bar seems like a good way to handle messages too (just ensuring that it's themed right, as it's light theme in this example). @jpnurmi what do you think?

**Comment by jpnurmi** [comment]
> I do think @madsrh seems ideal if that's possible. The snack bar seems like a good way to handle messages too (just ensuring that it's themed right, as it's light theme in this example). @jpnurmi what do you think?

Looks great and solves the issue with the red error color in the dark theme. What about the light theme, are we going to invert the colors? We should aim to fix the error color, of course, but I'm just wondering because currently, the issue is only visible to those who change to the dark theme. Inverting the colors would bring the issue visible by default.

**Comment by elioqoshi** [comment]
Regarding color: We need different shades for the same semantic color in light and dark. So on dark we need a different red than on light.

**Comment by elioqoshi** [comment]
Really checking on details here, but would it be possible to alter the page indicator active dot to be bigger than the others? I assume we are using the page indicator from Yaru right now so don't want to introduce too much conflict, though I think this creates a better distinction of where the user is.

![grafik](https://user-images.githubusercontent.com/5436686/222357041-7d836b65-500e-476c-9dae-f8b6d1bb3e74.png)

**Comment by madsrh** [comment]
> ...but would it be possible to alter the page indicator active dot to be bigger than the others?

Let's me summon @Jupi007 🧙

**Comment by jpnurmi** [comment]
It should be doable with the new theming capabilities @Jupi007 added. The dots all have the same size (which is good to keep the layout stable) but you can make the current dot visually larger by using an opaque decoration border, and transparent for the non-active ones.

- https://github.com/ubuntu/yaru_widgets.dart/pull/632

**Comment by Feichtmeier** [comment]
Altering the look is fine, as long as you keep the *default* look to look different enough from our radio buttons :radio_button:

**Comment by elioqoshi** [comment]
Sorry I pasted in a wrong screenshot. Updated it.

> The dots all have the same size (which is good to keep the layout stable) but you can probably make the current dot visually larger by using an opaque decoration border, and transparent for the non-active ones, for example.

Alternatively, the inactive dots could have a stroke in the same color as the background. Not sure if that is a hack or not though.

**Comment by d-loose** [comment]
Yes, will do!

**Comment by elioqoshi** [comment]
Thanks!

In my example I used 12px for the active dot and 8px for the inactive ones.

**Comment by Jupi007** [comment]
I just tried in the example of yaru_widgets, and it is possible to trick around with a border to change the item size, but it looks ugly because the transition white/transparent->orange is different from the grey->orange one.

[Capture vidéo du 2023-03-02 10-50-58.webm](https://user-images.githubusercontent.com/36476595/222393600-5bf17edb-ece5-4db1-b37c-80470eb8eb11.webm)

[Capture vidéo du 2023-03-02 10-53-15.webm](https://user-images.githubusercontent.com/36476595/222394140-465790ae-3ce9-4ae1-bba9-e01b53875169.webm)

It is funny, but I was working yesterday on some improvements of the page indicator (for which I requested a dev branch), and I added a `sizeBuilder`, in the same way as the `decorationBuilder`:

[Capture vidéo du 2023-03-02 10-55-25.webm](https://user-images.githubusercontent.com/36476595/222395274-65006733-955d-4228-bcb9-938339972b27.webm)

**Comment by d-loose** [comment]
I think it's easiest with yet another change in the yaru widgets. Rather than making the selected dot larger, I'd shrink all others and center them in a `SizedBox` which is as large as the selected one.
Works smoothly without any dots moving around:

[Screencast from 2023-03-02 11-41-01.webm](https://user-images.githubusercontent.com/113362648/222405876-b370a935-f10f-43a1-86fd-ece48a28e930.webm)
